framework/src/transfers/MultiAppReporterTransfer.C

```diff
@@ -133,15 +148,58 @@ MultiAppReporterTransfer::executeFromMultiapp()
   if (!hasFromMultiApp())
     return;
 
-  unsigned int ind = _subapp_index != std::numeric_limits<unsigned int>::max() ? _subapp_index : 0;
-  if (getFromMultiApp()->hasLocalApp(ind) &&
-      (!hasToMultiApp() || getToMultiApp()->hasLocalApp(ind)))
-    for (unsigned int n = 0; n < _from_reporter_names.size(); ++n)
-      transferReporter(_from_reporter_names[n],
-                       _to_reporter_names[n],
-                       getFromMultiApp()->appProblemBase(ind),
+  std::vector<unsigned int> indices;
+  if (_subapp_index == std::numeric_limits<unsigned int>::max() || _distribute_reporter_vector)
+  {
+    indices.resize(getFromMultiApp()->numGlobalApps());
+    std::iota(indices.begin(), indices.end(), 0);
+  }
+  else
+    indices = {_subapp_index};
+
+  if (_distribute_reporter_vector)
+    for (const auto & ind : indices)
+      for (unsigned int n = 0; n < _to_reporter_names.size(); ++n)
+      {
+        auto size = getFromMultiApp()->numGlobalApps();
+        clearVectorReporter(_to_reporter_names[n],
+                            hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
+                                            : getFromMultiApp()->problemBase());
+        resizeReporter(_to_reporter_names[n],
                        hasToMultiApp() ? getToMultiApp()->appProblemBase(ind)
-                                       : getFromMultiApp()->problemBase());
+                                       : getFromMultiApp()->problemBase(),
+                       size);
+      }
+
```

Comment:
Similar to the vectorpostprocessor transfer, we clear the reporter. Then we can
add to the reporter, and sum to get the entire vector on every proc.

Reply:
You should add these comments into the code If they are important for someone else to understand it.